netbox/dcim/models/device_components.py

```diff
@@ -1265,6 +1265,9 @@ class Meta:
     def get_absolute_url(self):
         return reverse('dcim:inventoryitem', kwargs={'pk': self.pk})
 
+    def get_device_status_color(self):
```

Comment:
We probably want to avoid introducing "proxy" methods that only exist to return the value of an attribute of some related object. Instead, maybe we can refactor `ChoiceFieldColumn` to accept a custom function for determining the appropriate color. Then we could do something like this:

```
device_status = columns.ChoiceFieldColumn(
    accessor=tables.A('device__status'),
    verbose_name=_('Device Status'),
    color=lambda x: x.device.get_status_color()
)
```

Reply:
I like where this is going. Let me know if you think the preference order I've got here looks right to you -- it takes the `color` callable first if provided, then falls back to the try-except of `get_FOO_color` as before.